db/background_mgr_attachment_migration.go

```diff
@@ -0,0 +1,351 @@
+	a.lock.Lock()
+	defer a.lock.Unlock()
+
+	a.CollectionIDs = collectionID
+}
+
+func (a *AttachmentMigrationManager) ResetStatus() {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+
+	a.DocsProcessed.Set(0)
+	a.DocsChanged.Set(0)
+	a.CollectionIDs = nil
+}
+
+func (a *AttachmentMigrationManager) GetProcessStatus(status BackgroundManagerStatus) ([]byte, []byte, error) {
+	a.lock.RLock()
+	defer a.lock.RUnlock()
+
+	response := AttachmentMigrationManagerResponse{
+		BackgroundManagerStatus: status,
+		MigrationID:             a.MigrationID,
+		DocsChanged:             a.DocsChanged.Value(),
+		DocsProcessed:           a.DocsProcessed.Value(),
+	}
+
+	meta := AttachmentMigrationMeta{
+		CollectionIDs: a.CollectionIDs,
+	}
+
+	statusJSON, err := base.JSONMarshal(response)
+	if err != nil {
+		return nil, nil, err
+	}
+	metaJSON, err := base.JSONMarshal(meta)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return statusJSON, metaJSON, err
+}
+
+func getMigrationDCPClientOptions(collectionIDs []uint32, groupID, prefix string) *base.DCPClientOptions {
+	checkpointPrefix := prefix + "att_migration:"
+	clientOptions := &base.DCPClientOptions{
+		OneShot:           true,
+		FailOnRollback:    false,
+		MetadataStoreType: base.DCPMetadataStoreCS,
+		GroupID:           groupID,
+		CollectionIDs:     collectionIDs,
+		CheckpointPrefix:  checkpointPrefix,
+	}
+	return clientOptions
+}
+
+type AttachmentMigrationManagerResponse struct {
+	BackgroundManagerStatus
+	MigrationID   string `json:"migration_id"`
+	DocsChanged   int64  `json:"docs_changed"`
+	DocsProcessed int64  `json:"docs_processed"`
+}
+
+type AttachmentMigrationMeta struct {
+	CollectionIDs []uint32 `json:"collection_ids"`
+}
+
+type AttachmentMigrationManagerStatusDoc struct {
+	AttachmentMigrationManagerResponse `json:"status"`
+	AttachmentMigrationMeta            `json:"meta"`
+}
+
+// GenerateAttachmentMigrationDCPStreamName returns the DCP stream name for a resync.
+func GenerateAttachmentMigrationDCPStreamName(migrationID string) string {
+	return fmt.Sprintf(
+		"sg-%v:att_migration:%v",
+		base.ProductAPIVersion,
+		migrationID)
+}
+
+// resetDCPMetadataIfNeeded will check for mismatch between current collectionIDs and collectionIDs on previous run
+func (a *AttachmentMigrationManager) resetDCPMetadataIfNeeded(ctx context.Context, datastore base.DataStore, database *Database, metadataKeyPrefix string, collectionIDs []uint32) error {
+	// if we are on our first run, no collections will be defined on the manager yet
+	if len(a.CollectionIDs) == 0 {
+		return nil
+	}
+	if len(a.CollectionIDs) != len(collectionIDs) {
+		err := a.PurgeDCPMetadata(ctx, datastore, database, metadataKeyPrefix)
+		if err != nil {
+			return err
+		}
+	}
+	slices.Sort(collectionIDs)
+	slices.Sort(a.CollectionIDs)
+
+	var purgeNeeded bool
+	for i, v := range collectionIDs {
+		if v != a.CollectionIDs[i] {
+			purgeNeeded = true
+			break
+		}
+	}
```

Comment:
possible tweak for readability:


```suggestion
	slices.Sort(collectionIDs)
	slices.Sort(a.CollectionIDs)

	purgeNeeded := !slices.Compare(collectionIDs, a.CollectionIDs) 
```